### PR TITLE
Upstream google

### PR DESCRIPTION
**Pull request by tartavull**
Integrates backs google/neuroglancer changes.

TODO: 
* State server -> checked from client to server
* check segmentation is being displayed correctly -> checked
* signals are being propagated -> checked
* check semantic mode works -> checked
* Merging and splitting signals -> checked
* Shattered -> checked
* Skeletons <- Not checked:
requires extending SkeletonSource from python/neuroglancer/skeleton.py
* Synapses <- checked
* Points <- checked
* Vector fields < Not checked:
The only backend that supports it seems to be https://github.com/saalfeldlab/render

Changes visible to the user:
 * It's now possible to overlap points, vector fields, and arbitrary 2D polygons on top of imaging and/or segmentation data.
 * Functionality to support multiple comma separated URLs to allow for hostname sharding.
 * Better support for skeletons
 * Support for pre-fetching chunks outside the viewport
 * Many bug fixes, including memory growing without bound which would crash the tab.

### CODE REVIEW

**Comment by william-silversmith** [comment]
This is very exciting!

File: python/example.py [hunk 1]
```diff
@@ -1,37 +1,57 @@
...
-#   name='segmentation', 
-#   voxel_size=[6, 6, 40], 
-#   graph='./snemi3d/snemi3d_graph.pickle'
-# )
-
-url = viewer.get_viewer_url()
-webbrowser.open(url)
-print(url)
-
-if not sys.flags.interactive:
```
Comment (william-silversmith) [inline python/example.py]: We shouldn't overwrite this new logic here. This should be merged in. It allows you to run the server with `python example.py` without `-i`

**Comment by jingpengw** [comment]
exciting! I see skeleton support!

File: python/example.py (hunk 1 above)
Comment (tartavull) [reply python/example.py]: good catch, added back.

**Comment by tartavull** [comment]
Sorry, no skeletonization code! we need yours @jingpengw

File: python/example.py [hunk 2]
```diff
@@ -1,37 +1,62 @@
...
-import h5py
-import webbrowser
-import neuroglancer
-from neuroglancer import static
 import sys
+import webbrowser
 
-path = './neuroglancer/static/'
-
-neuroglancer.set_static_content_source(path=path)
```
Comment (william-silversmith) [inline python/example.py]: Changing this line broke it for me.

**Comment by william-silversmith** [comment]
It looks like drawing lines still doesn't work on OS X. We'll need to fix that at some point. :(

File: python/example.py (hunk 2 above)
Comment (tartavull) [reply python/example.py]: Is a good thing I deleted it , right?

**Pull request merged**